### PR TITLE
Stop using vault dev

### PR DESCRIPTION
**Pull request by bhechinger**
We now deploy a proper vault and go through all the steps to initialize and unseal it.

### CODE REVIEW

File: app/tooling/sales-admin/commands/vault_init.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		initResponseJson, err := os.ReadFile(credentialsFileName)
+		if err != nil {
+			log.Fatalf("Error reading %s file: %s", credentialsFileName, err)
+		}
+
+		err = json.Unmarshal(initResponseJson, &initResponse)
+		if err != nil {
+			log.Fatalf("Error Unmarshalling json: %s", err)
+		}
+	} else if errors.Is(err, os.ErrNotExist) {
```
Comment (ardan-bkennedy) [inline app/tooling/sales-admin/commands/vault_init.go:45]: Don't use an else clause unless you have ABSOLUTELY no choice. A switch is a better option.

File: foundation/vault/vault.go [hunk 2]
```diff
@@ -36,6 +36,25 @@ var defaultClient = http.Client{
 	},
 }
 
+type InitRequest struct {
```
Comment (ardan-bkennedy) [inline foundation/vault/vault.go:39]: Let's add a modules.go file in this package for these types.

File: foundation/vault/vault.go [hunk 3]
```diff
@@ -150,6 +172,151 @@ func (v *Vault) PublicKeyPEM(kid string) (string, error) {
...
+		return nil, fmt.Errorf("do: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status code: %s", resp.Status)
+	}
+
+	var response InitResponse
+
```
Comment (ardan-bkennedy) [inline foundation/vault/vault.go:204]: remove this line space and keep the variable declaration agaisnt the code that is using it.

File: foundation/vault/vault.go [hunk 4]
```diff
@@ -150,6 +172,151 @@ func (v *Vault) PublicKeyPEM(kid string) (string, error) {
...
+	return nil
+}
+
+func (v *Vault) Mount(ctx context.Context, opts *MountInput) error {
+	mounts, err := v.ListMounts(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting mount list: %w", err)
+	}
+
+	// Mount already exists so we'll do nothing
```
Comment (ardan-bkennedy) [inline foundation/vault/vault.go:249]: Comments must have proper sentence structure including punctuation.

Comment (ardan-bkennedy) [reply foundation/vault/vault.go:249]: Apply this everywhere.

File: app/tooling/sales-admin/commands/vault_init.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	"log"
+	"os"
+	"strings"
+
+	"github.com/ardanlabs/service/foundation/vault"
+)
+
+const credentialsFileName = "/tmp/vault-init.json"
+
+// VaultInit sets up a newly provisioned vault instance
```
Comment (ardan-bkennedy) [inline app/tooling/sales-admin/commands/vault_init.go:17]: Said this later. All comments must have proper sentence structure, punctuation, and grammar

File: app/tooling/sales-admin/commands/vault_init.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+func VaultInit(vaultConfig vault.Config) error {
+	var (
```
Comment (ardan-bkennedy) [inline app/tooling/sales-admin/commands/vault_init.go:19]: Not a fan of var blocks, especially inside a function.  I rather have these separated

var initResponse vault.InitResponse

Let's not use pointer semantics for variable declaration and use var for zero value construction.

Comment (ardan-bkennedy) [reply app/tooling/sales-admin/commands/vault_init.go:19]: Noticing this is not needed until line 41, so let's move it.

Comment (ardan-bkennedy) [reply app/tooling/sales-admin/commands/vault_init.go:19]: Declare a variable as close to where it is needed.

File: app/tooling/sales-admin/main.go [hunk 7]
```diff
@@ -29,12 +29,8 @@ type config struct {
 	}
 	Vault struct {
 		KeysFolder string `conf:"default:zarf/keys/"`
-		Address    string `conf:"default:http://vault-service.sales-system.svc.cluster.local:8200"`
+		Address    string `conf:"default:http://vault.sales-system.svc.cluster.local:8200"`
```
Comment (ardan-bkennedy) [inline app/tooling/sales-admin/main.go:32]: Interested to learn more about this.

**Comment by veqqq** [comment]
After watching Bill's keynote talk: https://www.youtube.com/watch?v=IV0wrVb31Pg I decided to check this out. (Unfortunately, checking out to the 2018 version, most dependencies are no longer available so the cool local logging approach isn't quite observable.) It isn't working so well, most likely because this is a course so I'm probably missing relevant paid material.

Running `make test`, as the makefile suggests, fails trying to access this non-existent folder:  `github.com/ardanlabs/service/foundation/vault`

**Pull request merged**